### DISCUSSION TITLE
Recommended PHP coroutines library with Workerman

### DISCUSSION THREAD

**Discussion by cayolblake** [walkor/Workerman#971] [Q&A]
Hello @walkor 

What's a recommended PHP coroutines library to work with Workerman?

**Comment by walkor** [comment]
swoole

**Comment by cayolblake** [comment]
Since Swoole provides similar APIs to what Workerman provides, I am trying to understand when/why to use Workerman while setting the event loop to Swoole? I mean what are the benefits if I'm going to use Swoole anyway?

**Reply by walkor** [reply]
If you need collaboration, using Swoole directly is a better choice
